Mathlib/LinearAlgebra/TensorAlgebra/Hopf.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2025 Nikolas Tapia. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nikolas Tapia
+-/
+module
+
+public import Mathlib.LinearAlgebra.TensorAlgebra.Basic
+public import Mathlib.RingTheory.Coalgebra.Convolution
+public import Mathlib.RingTheory.HopfAlgebra.Basic
+
+/-!
+# Hopf algebra structure on `TensorAlgebra R M`
+
+In this file we implement the natura Hopf algebra structure on `TensorAlgebra R M`.
+The comultiplication is the unique algebra map satisfying `comul ((ι R) x) = (ι R) x ⊗ 1 + 1 ⊗ (ι R)
+x` for all `x : M`.
+`algebraMapInv` acts as the counit, and the antipode is the unique algebra map `antipode :
+TensorAlgebra R M → (TensorAlgebra R M)ᵐᵒᵖ` induced by `fun x => op -(ι R) x`.
+-/
+@[expose] public section
+
+namespace TensorAlgebra
+open scoped TensorProduct RingTheory.LinearMap Coalgebra
+open LinearMap TensorProduct
+
+variable (R : Type*) [CommRing R] {M : Type*} [AddCommMonoid M] [Module R M]
+
+local notation "T["M"]" => TensorAlgebra R M
+
+/-- Linear map inducing the comultiplication in `TensorAlgebra R M`. -/
+def comul' : M →ₗ[R] T[M] ⊗[R] T[M] :=
+  ((ι R) ⊗ₘ (Algebra.linearMap R T[M])) ∘ₗ (TensorProduct.rid R M).symm +
+  ((Algebra.linearMap R T[M]) ⊗ₘ (ι R)) ∘ₗ (TensorProduct.lid R M).symm
```

Comment:
This can be inlined (i.e. subsumed) in the next definition, since every time you use it is via `comul` anyway.